tests/jupyterhub_api/test_token.py

```diff
@@ -0,0 +1,92 @@
+
+import json
+import requests
+import subprocess
+import os
+import docker
+import sys
+sys.path.append("..")
+import yaml
+import time
+from logger import Logger, LOG_FOLDER, LOG_EXTENSION
+import argparse
+from SessionUtils import Test, JupyterhubTest
+from requests.packages.urllib3.exceptions import InsecureRequestWarning
+requests.packages.urllib3.disable_warnings(InsecureRequestWarning)
+
+def get_args():
+    parser = argparse.ArgumentParser(description='Arguments', formatter_class = argparse.ArgumentDefaultsHelpFormatter)
+    parser.add_argument("--port", dest="port", type=int,
+                        required = True,
+                        )
+    parser.add_argument( "--users", nargs='+', dest="users",
+                        required  = True,
+                        help='list of users')
+
+    parser.add_argument("--token", dest="token",
+                        required=False,
+                        help='token')
+
+    parser.add_argument("--base_path", dest="base",
+                        required=False,
+                        help='base path')
+
+    args = parser.parse_args()
+    return args
+
+
+class Token(JupyterhubTest):
+    """
+    Test the vailidity of token
+
+    To run the test use the following command:
+    python3 test_token.py --port 443 --users user1 --base_path ""
+
+    For multiple users
+
+    python3 test_token.py --port 443 --users user0 user1 user2 --base_path ""
+
+    """
+    def __init__(self, port, token, users, base_path, verify):
+        param = {}
+        param['test_name'] = "Check_Token"
+        JupyterhubTest.__init__(self, port, token, base_path, verify, **param)
+        self.users = users
+
+    def check_user(self):
+
+            """
+            Submit a  get request to  https://localhost:443/hub/api/users/user{} to check if a token is valid or not
+
+            """
+            for user in self.users:
+                try:
+                    r = super().api_calls("get", user, endpoint ="")
+
+                except requests.exceptions.RequestException as e:
+                    self.log.write("error", str(e))
+                    self.exit = 1
+                else:
+                    if (r.status_code != 403):
```

Comment:
Could it be that you get 401-Unauthorized in some cases?

Reply:
I have tried using invalid tokens many times but didn't get 401 status code.